ibm/resource_ibm_cbr_rule.go

```diff
@@ -0,0 +1,527 @@
+							Description: "The attributes.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": &schema.Schema{
+										Type:        schema.TypeString,
+										Required:    true,
+										Description: "The attribute name.",
+									},
+									"value": &schema.Schema{
+										Type:        schema.TypeString,
+										Required:    true,
+										Description: "The attribute value.",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"resources": &schema.Schema{
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "The resources this rule apply to.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"attributes": &schema.Schema{
+							Type:        schema.TypeList,
+							Required:    true,
+							Description: "The resource attributes.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": &schema.Schema{
+										Type:        schema.TypeString,
+										Required:    true,
+										Description: "The attribute name.",
+									},
+									"value": &schema.Schema{
+										Type:        schema.TypeString,
+										Required:    true,
+										Description: "The attribute value.",
+									},
+									"operator": &schema.Schema{
+										Type:        schema.TypeString,
+										Optional:    true,
+										Description: "The attribute operator.",
+									},
+								},
+							},
+						},
+						"tags": &schema.Schema{
+							Type:        schema.TypeList,
+							Optional:    true,
+							Description: "The optional resource tags.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": &schema.Schema{
+										Type:        schema.TypeString,
+										Required:    true,
+										Description: "The tag attribute name.",
+									},
+									"value": &schema.Schema{
+										Type:        schema.TypeString,
+										Required:    true,
+										Description: "The tag attribute value.",
+									},
+									"operator": &schema.Schema{
+										Type:        schema.TypeString,
+										Optional:    true,
+										Description: "The attribute operator.",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"x_correlation_id": &schema.Schema{
+				Type:         schema.TypeString,
+				Computed:     true,
+				ValidateFunc: InvokeValidator("ibm_cbr_rule", "x_correlation_id"),
+				Description:  "The supplied or generated value of this header is logged for a request and repeated in a response header for the corresponding response. The same value is used for downstream requests and retries of those requests. If a value of this headers is not supplied in a request, the service generates a random (version 4) UUID.",
+			},
+			"transaction_id": &schema.Schema{
+				Type:         schema.TypeString,
+				Computed:     true,
+				ValidateFunc: InvokeValidator("ibm_cbr_rule", "transaction_id"),
```

Comment:
computed attributes are API responses..
Validations is not required for computed attributes

Reply:
removed